src/azure-api-utils.ts

```diff
@@ -1,9 +1,17 @@
-export function parseResource(armId: string): { resourceGroupName: string | undefined; name: string | undefined } {
+export function parseResource(armId: string): {
+    parentResourceId: string | undefined;
+    subscriptionId: string | undefined;
+    resourceGroupName: string | undefined;
+    name: string | undefined;
+} {
+    // {parentResourceId}/members/{name}
     // /subscriptions/{subid}/resourcegroups/{group}/providers/.../{name}
     const bits = armId.split("/").filter((bit) => bit.length > 0);
     const resourceGroupName = bits[3];
+    const subscriptionId = bits[1];
     const name = bits[bits.length - 1];
-    return { resourceGroupName, name };
+    const parentResourceId = `/${bits.slice(0, bits.length - 2).join("/")}`;
+    return { parentResourceId, subscriptionId, resourceGroupName, name };
 }
 
 export function parseSubId(armId: string): { subId: string } {
```

Comment:
this func was added in your PR @Tatsinnit 
follow-up refactoring might be to delete it, and reuse the parseResource that allows extracting more parts.